packages/in_app_purchase/ios/Classes/InAppPurchasePlugin.m

```diff
@@ -31,4 +40,26 @@ - (void)canMakePayments:(FlutterResult)result {
   result([NSNumber numberWithBool:[SKPaymentQueue canMakePayments]]);
 }
 
+- (void)getProductListWithMethodCall:(FlutterMethodCall *)call result:(FlutterResult)result {
+  NSArray *productsIdentifiers = call.arguments[@"identifiers"];
+  [self.productRequestHandler
+      startWithProductIdentifiers:[NSSet setWithArray:productsIdentifiers]
+                completionHandler:^(SKProductsResponse *_Nullable response) {
+                  NSMutableArray *productsDetailsJSON = [NSMutableArray new];
```

Comment:
nit: `productsDetailsJSON` is a confusing name because this isn't a JSON string, it's just an array of dictionaries. `productDetailsSerialized`?